apiextensions/v1alpha1/types_compatibilityrequirement.go

```diff
@@ -0,0 +1,391 @@
+	CompatibilityRequirementCompatibleWithWarningsReason string = "CompatibleWithWarnings"
+
+	// CompatibilityRequirementCompatibleReason surfaces when the CRD exists and is compatible with this requirement.
+	CompatibilityRequirementCompatibleReason string = "Compatible"
+)
+
+// CompatibilityRequirementStatus defines the observed status of the Compatibility Requirement.
+// +kubebuilder:validation:MinProperties=1
+type CompatibilityRequirementStatus struct {
+	// conditions is a list of conditions and their status.
+	// Known condition types are Progressing, Admitted, Compatible.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=32
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// observedCRD documents the uid and generation of the CRD object when the current status was written.
+	// This field will be omitted if the target CRD does not exist or could not be retrieved.
+	// +optional
+	ObservedCRD ObservedCRD `json:"observedCRD,omitzero"`
+
+	// crdName is the name of the target CRD. The target CRD is not required to
+	// exist, as we may legitimately place requirements on it before it is
+	// created.  The observed CRD is given in status.observedCRD, which will be
+	// empty if no CRD is observed.
+	// When present, must be between 1 and 253 characters and conform to RFC 1123 subdomain format:
+	// lowercase alphanumeric characters, '-' or '.', starting and ending with alphanumeric characters.
+	// When not specified, the requirement applies to any CRD name discovered from the compatibility schema.
+	// This field is optional. Once set, the value cannot be changed and must always remain set.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="crdName cannot be changed once set"
+	// +optional
+	CRDName string `json:"crdName,omitempty"`
+}
+
+// ObservedCRD contains information about the observed target CRD.
+// +kubebuilder:validation:MinProperties=1
+type ObservedCRD struct {
+	// uid is the uid of the observed CRD.
+	// Must be a valid UUID consisting of lowercase hexadecimal digits in 5 hyphenated blocks (8-4-4-4-12 format).
+	// Length must be between 1 and 36 characters.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=36
+	// +kubebuilder:validation:Format=uuid
+	// +required
+	UID string `json:"uid,omitempty"`
+
+	// generation is the observed generation of the CRD.
```

Comment:
Generation should monotonically increase right? Validate that with CEL

Reply:
It doesn't have to increase monotonically because it can reset if uid changes. I don't think that's too onerous to write, though. I'll add it.